pkg/dnfjson/dnfjson.go

```diff
@@ -722,10 +722,12 @@ type transactionArgs struct {
 }
 
 type packageSpecs []PackageSpec
+type moduleSpecs map[string]ModuleSpec
 
 type depsolveResult struct {
 	Packages packageSpecs          `json:"packages"`
 	Repos    map[string]repoConfig `json:"repos"`
+	Modules  moduleSpecs           `json:"modules"`
```

Comment:
Nitpick: could probably be marked as `omitempty`.

Reply:
I thought `omitempty` only applies when encoding, when decoding `moduleSpecs` is already initialized to its zero value if not present?

Reply:
Correct. Although functionally irrelevant, it also expresses that it is optional and may not be present. I'm OK with keeping it as is.